### PR TITLE
refactor: refactor the directory organization of procedure

### PR DESCRIPTION
**Pull request by ZuLiangWang**
# Which issue does this PR close?

Closes #

# Rationale for this change
We have developed multiple Procedures to manage the state machine and retry process, but they are defined under the same `procedure` package. In order to avoid constant conflicts, we have adopted unreasonable naming methods. Now do a refactoring and disassemble them. Divide into multiple packages, and adjust the naming of constants.

# What changes are included in this PR?
* Split procedures to multiple packages.
* Rename constants.

# Are there any user-facing changes?
None.

# How does this change test
Passing exists unit tests.

### CODE REVIEW

**Comment by chunshao90** [comment]
Abbreviate `partition` as `par`.
Abbreviate `request` as `req`.
Abbreviate `response` as `resp`.

File: server/coordinator/procedure/dml/createpartitiontable/create_partition_table.go [hunk 1]
```diff
@@ -69,53 +70,53 @@ type CreatePartitionTableProcedure struct {
...
+	Storage              procedure.Storage
+	Req                  *metaservicepb.CreateTableRequest
+	PartitionTableShards []cluster.ShardNodeWithVersion
+	SubTablesShards      []cluster.ShardNodeWithVersion
+	OnSucceeded          func(cluster.CreateTableResult) error
+	OnFailed             func(error) error
 }
 
-func NewCreatePartitionTableProcedure(request CreatePartitionTableProcedureRequest) *CreatePartitionTableProcedure {
+func NewCreatePartitionTableProcedure(request ProcedureRequest) *Procedure {
```
Comment (chunshao90) [inline server/coordinator/procedure/dml/createpartitiontable/create_partition_table.go:88]: NewProcedure?

File: server/coordinator/procedure/dml/createtable/create_table.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+	Dispatch    eventdispatch.Dispatch
+	Cluster     *cluster.Cluster
+	ID          uint64
+	ShardID     storage.ShardID
+	Req         *metaservicepb.CreateTableRequest
+	OnSucceeded func(cluster.CreateTableResult) error
+	OnFailed    func(error) error
+}
+
+func NewCreateTableProcedure(request ProcedureRequest) procedure.Procedure {
```
Comment (chunshao90) [inline server/coordinator/procedure/dml/createtable/create_table.go:113]: NewProcedure?

**Comment by jiacai2050** [comment]
https://www.allacronyms.com/partition/abbreviated

part is more prefer than par

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**